test/provision/compile.sh

```diff
@@ -25,7 +26,7 @@ then
         echo "Not on master K8S node; no need to compile Cilium container"
     fi
 else
-    make
+    sudo -u vagrant -H -E make CILIUM_USE_ENVOY=1 PKG_BUILD=1
```

Comment:
I have not seen this pattern before (sudo -u vagrant). Do we need it here to make the build run by user "vagrant"?

Also, PKG_BUILD=1 is so far only used for container builds, where the build is done as root. envoy/Makefile will build the release version if PKG_BUILD is defined. Not sure we want it like that. This should work when not in a container:

```make CILIUM_USE_ENVOY=1```

Reply:
If the build is done as root you should add ```PKG_BUILD=1```